### PR TITLE
Use row_count instead of batch and emit py:decimal from fetchone

### PR DESCRIPTION
**Pull request by pdet**
In this pr, I remove the necessity of checking batch_sizes and use the row_count from the arrow table to perform cardinality estimation and to use progress bars.
It also adds a test for a big decimal, and changes the fetchone method to emit python decimals instead of doubles.

### CODE REVIEW

File: src/function/table/arrow.cpp [hunk 1]
```diff
@@ -294,8 +294,7 @@ void ArrowTableFunction::ArrowScanFunctionParallel(ClientContext &context, const
 }
 
 idx_t ArrowTableFunction::ArrowScanMaxThreads(ClientContext &context, const FunctionData *bind_data_p) {
-	auto &data = (const ArrowScanFunctionData &)*bind_data_p;
-	return data.stream->number_of_batches;
+	return context.db->NumberOfThreads();
```
Comment (Mytherin) [inline src/function/table/arrow.cpp:297]: The stream still has a number of rows, right? Can we use that as a proxy here if it is set, e.g. one thread every 100K rows?

**Review by Mytherin — COMMENTED** [review]
Thanks! Looks good. Minor comment:

File: src/function/table/arrow.cpp [hunk 2]
```diff
@@ -294,7 +294,9 @@ void ArrowTableFunction::ArrowScanFunctionParallel(ClientContext &context, const
 }
 
 idx_t ArrowTableFunction::ArrowScanMaxThreads(ClientContext &context, const FunctionData *bind_data_p) {
-	return context.db->NumberOfThreads();
+	auto &bind_data = (const ArrowScanFunctionData &)*bind_data_p;
+	idx_t rows_per_thread = 100000;
```
Comment (hannesmuehleisen) [inline src/function/table/arrow.cpp:298]: Hardcoded constant?

Comment (pdet) [reply src/function/table/arrow.cpp:298]: I can also pragma it up.
Or is there another preferred way?

Comment (hannesmuehleisen) [reply src/function/table/arrow.cpp:298]: How about an optional parameter to the arrow scan function?

Comment (pdet) [reply src/function/table/arrow.cpp:298]: sure, that sounds good, will do it later today :-)

**Pull request merged**